packages/graph/hash_graph/lib/graph/src/store/query/filter.rs

```diff
@@ -0,0 +1,327 @@
+use std::{
+    borrow::Cow,
+    fmt::{Debug, Formatter},
+};
+
+use serde::Deserialize;
+
+use crate::store::query::{Expression, Literal, Path, QueryRecord};
+
+/// A set of conditions used for queries.
+#[derive(Deserialize)]
+#[serde(
+    rename_all = "camelCase",
+    bound = "'de: 'q, T::Path<'q>: Deserialize<'de>"
+)]
+pub enum Filter<'q, T: QueryRecord> {
+    All(Vec<Self>),
```

Comment:
I am not sure we've talked through quantifiers vs operators
How would `And` and `Or` instead of `All` and `Any` look like? Would it be easier to transpile?

For example, we we have the following filter:
`A = B && (Not (C = A || C = D) || A == D) && D = F
It would be the following with `All`/`Any`:
```
All[
  A = B,
  Any[
    Not (
      Any[
        C = A,
        C = D
      ]
    ),
    A == D
  ]
  D = F
]
```
And would be this for `And`/`Or` (with only 2 expressions `(l, r)` instead of `[e1, e2, .., en]`):
```
And(
  And(
    A = B
    Or(
      Not(
        Or(
          C = A,
          C = D
        ),
      )  
      A = D
    )
  )
  D = F
)
```

Are you running into annoyances with `All`` and `Any` when transpiling SQL? Do we think there might be some burden (for other devs) working with 'quantifiers' when most other things would be using `And` and `Or`? Not saying we have to change, but just want to make sure I understand :)

Reply:
I suppose the original intention was that we could apply the filters on the same level in "one go" (using iterator `all` and `any` methods?) when we stream each entry, is that the intended benefit?